packages/pipeline-editor/src/index.ts

```diff
@@ -170,18 +223,43 @@ const extension: JupyterFrontEndPlugin<void> = {
       category: 'Elyra'
     });
 
-    // Add the command to the launcher
-    if (launcher) {
-      launcher.add({
-        command: openPipelineEditorCommand,
-        category: 'Elyra',
-        rank: 1
-      });
-    }
-    // Add new pipeline to the file menu
-    menu.fileMenu.newMenu.addGroup(
-      [{ command: openPipelineEditorCommand }],
-      30
+    PipelineService.getRuntimesSchema().then(
+      (schema: any) => {
+        // Add the command to the launcher
+        if (launcher) {
+          launcher.add({
+            command: openPipelineEditorCommand,
+            category: 'Elyra',
+            rank: 1
+          });
+          for (const runtime of schema) {
+            launcher.add({
+              command: openPipelineEditorCommand,
+              category: 'Elyra',
+              args: { runtime },
+              rank:
+                runtime.name === 'kfp' ? 2 : runtime.name === 'airflow' ? 3 : 4
+            });
+          }
+        }
+        // Add new pipeline to the file menu
+        menu.fileMenu.newMenu.addGroup(
+          [{ command: openPipelineEditorCommand }],
+          30
+        );
+        for (const runtime of schema) {
+          menu.fileMenu.newMenu.addGroup(
+            [
+              {
+                command: openPipelineEditorCommand,
+                args: { runtime }
+              }
+            ],
+            runtime.name === 'kfp' ? 2 : runtime.name === 'airflow' ? 3 : 4
```

Comment:
For some reason the generic pipeline editor isn't following the specified order
![image](https://user-images.githubusercontent.com/25207344/116298147-170c8300-a76a-11eb-8504-7904711d658a.png)

Reply:
Slightly off-topic - in this case we should probably not rely on the icon to identify the typed editor. In this context menu the icons are really small and users might find it hard to distinguish them. Can we use `Kubeflow Pipelines Pipeline Editor` and `Apache Airflow Pipeline Editor` as text instead? We are already doing this for the script editors (: